eln/preference.js

```diff
@@ -0,0 +1,58 @@
+import API from 'src/util/api';
+import Roc from '../rest-on-couch/Roc';
+
+export async function preferencesFactory(id, options) {
+    const {
+        name = 'viewPreferences',
+        url = undefined,
+        database = 'eln',
+        initial = []
+    } = options;
+
+    const kind = 'viewPreferences';
+
+    var roc = new Roc({
+        url: url,
+        database: database,
+        kind: kind
+    });
+
+    var user = await roc.getUser();
+    id += '_' + user.username;
+
+    var existing = (await roc.view('entryByKindAndId', {key: [kind, id]}))[0];
+    var preferenceRoc;
+    var rocOptions = {
+        varName: id,
```

Comment:
Are you creating a variable with the email address in it's name? That seems weird, it will be different for every user. Choose something not too general to avoid variable name collisions. Like `elnPreferences`.